repo/manifest/manifest_manager.go

```diff
@@ -84,67 +81,59 @@ func (m *Manager) Put(ctx context.Context, labels map[string]string, payload int
 		Content: b,
 	}
 
+	m.mu.Lock()
 	m.pendingEntries[e.ID] = e
+	m.mu.Unlock()
 
 	return e.ID, nil
 }
 
 // GetMetadata returns metadata about provided manifest item or ErrNotFound if the item can't be found.
 func (m *Manager) GetMetadata(ctx context.Context, id ID) (*EntryMetadata, error) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
+	e, err := m.getPendingOrCommitted(ctx, id)
+	if err != nil {
+		return nil, err
+	}
 
-	e := m.pendingEntries[id]
-	if e == nil {
-		var err error
+	return cloneEntryMetadata(e), nil
```

Comment:
This is not the same as before.

This is assuming that either:
* `e` points to a local copy (which it does not); or
* `*e` is not modified in any way, which currently seems to be the case but it is subtle and there is not explicit guarantee

Reply:
entries (pending or committed) are immutable, so I don't really see a problem here.